docs/src/main/asciidoc/cdi-reference.adoc

```diff
@@ -931,6 +931,8 @@ public class Processor {
 <1> The injected instance is an _immutable list_ of the contextual references of the _disambiguated_ beans.
 <2> For this injection point the required type is `Service` and no additional qualifiers are declared.
 
+TIP: By default, the list of beans is sorted by priority as defined by `io.quarkus.arc.InjectableBean#getPriority()`. Higher priority goes first. In general, the `@javax.annotation.Priority` and `@io.quarkus.arc.Priority` annotations can be used to assign the priority to a class bean, producer method or producer field.
```

Comment:
> Higher priority goes first.

That is, _in descending order_?

Not bad in and of itself, but inconsistent with how interceptors are ordered for example.

Reply:
> That is, _in descending order_?

Yes.

> Not bad in and of itself, but inconsistent with how interceptors are ordered for example.

I think that interceptors are a bit different, because the interceptor with highest priority is "closest"  (around) to the original invocation. If you speak about observers then yes, it is inconsistent and the spec wording is IMO very unfortunate and does not make a lot of sense - _"Observers with smaller priority values are called first."_ (?). In other words, it's intentional and I believe that descending order makes a lot more sense for this particular use case.

Reply:
Yea observers are also like that. It seems to me that whenever CDI uses a priority to _order_ a sequence, it's always in an _ascending_ direction.

I agree these are all somewhat different use cases, and this is perhaps closest to how alternatives are selected, but still... better be sure this is the right way to do it :-)

Reply:
I'm not sure it's the "right way" but I'm pretty sure it makes sense ;-)

Reply:
This ordering makes sense. `services.get(0)` should logically give you the selected alternative.
It also puts all the beans without any priority (i.e. with 0) at the end of the list which is sensible.